### PR TITLE
Git Init in current Workspace

### PR DESCRIPTION
**Pull request by jbicker**
#### What it does
Fixes https://github.com/theia-ide/theia/issues/5936
The current Workspace can be initialized as a Git Repository now.
git-init

#### How to test
Create a new Workspace and open it in theia.
Open the git view. You should see the note that there is no repository yet.
Hit the plus in the Git View Header. 
Now you should see the Git View with the respective commit message field and so on.

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to review in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by akosyakov** [comment]
@jbicker please follow PR template: https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#opening-a-pull-request

File: packages/git/src/browser/git-quick-open-service.ts [hunk 1]
```diff
@@ -47,7 +49,8 @@ export class GitQuickOpenService {
         @inject(QuickOpenService) protected readonly quickOpenService: QuickOpenService,
         @inject(MessageService) protected readonly messageService: MessageService,
         @inject(WorkspaceService) protected readonly workspaceService: WorkspaceService,
-        @inject(FileSystem) protected readonly fileSystem: FileSystem
+        @inject(FileSystem) protected readonly fileSystem: FileSystem,
+        @inject(LabelProvider) protected readonly labelProvider: LabelProvider
```
Comment (vince-fugnitto) [inline packages/git/src/browser/git-quick-open-service.ts]: Can we refactor `labelProvider` as a property injection over a constructor injection.
Injecting new constructor injections is a breaking change based on our [coding guidelines](https://github.com/theia-ide/theia/wiki/Coding-Guidelines#property-injection)

File: packages/scm/src/browser/scm-widget.tsx [hunk 2]
```diff
@@ -94,7 +94,7 @@ export class ScmWidget extends ReactWidget implements StatefulWidget {
         this.toDisposeOnRefresh.dispose();
         this.toDispose.push(this.toDisposeOnRefresh);
         const repository = this.scmService.selectedRepository;
-        this.title.label = repository ? repository.provider.label : 'no repository opened';
+        this.title.label = repository ? repository.provider.label : '';
```
Comment (akosyakov) [inline packages/scm/src/browser/scm-widget.tsx]: it won't work well with multiple views in the scm view container. The header of scm widget part will be empty then.

One can do it by installing https://github.com/microsoft/vscode-extension-samples/tree/master/tree-view-sample from sources and changing its package.json that views are added to the scm view container.

File: packages/git/src/browser/git-contribution.ts [hunk 3]
```diff
@@ -456,19 +467,26 @@ export class GitContribution implements CommandContribution, MenuContribution, T
...
             const id = '__git.tabbar.toolbar.' + commandId;
             const command = this.commands.getCommand(commandId);
             this.commands.registerCommand({ id, iconClass: command && command.iconClass }, {
                 execute: (widget, ...args) => widget instanceof ScmWidget && this.commands.executeCommand(commandId, ...args),
                 isEnabled: (widget, ...args) => widget instanceof ScmWidget && this.commands.isEnabled(commandId, ...args),
-                isVisible: (widget, ...args) => widget instanceof ScmWidget && this.commands.isVisible(commandId, ...args),
+                isVisible: (widget, ...args) =>
+                    widget instanceof ScmWidget &&
+                    this.commands.isVisible(commandId, ...args) &&
+                    !!this.repositoryProvider.selectedRepository === whenRepositorySelected,
```
Comment (akosyakov) [inline packages/git/src/browser/git-contribution.ts]: I wonder can we use `when` closure on a command instead? for example `!scmProvider` for missing scm provider at all or `scmProvider == git` for cases when scm provider is git

Comment (akosyakov) [reply packages/git/src/browser/git-contribution.ts]: BTW where do you pass it as `false`?

Comment (jbicker) [reply packages/git/src/browser/git-contribution.ts]: Here https://github.com/theia-ide/theia/tree/8e4ceac6ff943c46586d62e25d7f847ff3f01c63#L485
When I look at it again, I think that it is total nonsense. I will change it, use the helper function as before just for the other repository related commands and register this Init Toolbar Item seperately.

File: packages/git/src/common/git.ts [hunk 4]
```diff
@@ -841,6 +841,12 @@ export interface Git extends Disposable {
     // tslint:disable-next-line:no-any
     lsFiles(repository: Repository, uri: string, options?: Git.Options.LsFiles): Promise<any>;
 
+    /**
+     * Initialises a new git repository in the folder with the given uri.
+     *
+     * @param uri the uri of the folder
+     */
+    initRepository(uri: string): Promise<string>;
```
Comment (kittaakos) [inline packages/git/src/common/git.ts]: You know, you can do `git.exec({ localUri: yourUri }, ['init'])`. No Git API changes are required.

File: packages/git/src/common/git.ts [hunk 5]
```diff
@@ -840,7 +840,6 @@ export interface Git extends Disposable {
      */
     // tslint:disable-next-line:no-any
     lsFiles(repository: Repository, uri: string, options?: Git.Options.LsFiles): Promise<any>;
```
Comment (akosyakov) [inline packages/git/src/common/git.ts]: this file can be reverted?

**Pull request merged**